tabular/src/autogluon/tabular/predictor/predictor.py

```diff
@@ -898,6 +926,164 @@ def fit_extra(self, hyperparameters, time_limit=None, base_model_names=None, **k
+        X_test = unlabeled_data.copy()
+
+        for i in range(max_iter):
+            iter_print = str(i + 1)
+            logger.log(20, f'Beginning iteration {iter_print} of pseudolabeling out of max: {max_iter}')
+            y_pred_proba = self.predict_proba(data=X_test, as_multiclass=True)
+            y_pred = get_pred_from_proba_df(y_pred_proba, problem_type=self.problem_type)
+
+            test_pseudo_idxes_true = filter_pseudo(y_pred_proba_og=y_pred_proba, problem_type=self.problem_type)
+
+            if return_pred_prob:
+                if i == 0:
+                    y_pred_proba_og = y_pred_proba
+                else:
+                    y_pred_proba_og.loc[test_pseudo_idxes_true.index] = y_pred_proba.loc[test_pseudo_idxes_true.index]
+
+            if len(test_pseudo_idxes_true) < 1:
+                logger.log(20,
+                           f'Could not confidently assign pseudolabels for any of the provided rows in iteration: {iter_print}. Done with pseudolabeling...')
+                break
+            else:
+                logger.log(20, f'Pseudolabeling algorithm confidently assigned pseudolabels to: {len(test_pseudo_idxes_true)} rows of data'
+                               f'on iteration: {iter_print}. Adding to train data')
+
+            test_pseudo_idxes = pd.Series(data=False, index=y_pred_proba.index)
+            test_pseudo_idxes[test_pseudo_idxes_true.index] = True
+
+            y_pseudo_og = y_pseudo_og.append(y_pred.loc[test_pseudo_idxes_true.index], verify_integrity=True)
+
+            pseudo_data = unlabeled_data.loc[y_pseudo_og.index]
+            pseudo_data[self.label] = y_pseudo_og
+            self.fit_extra(pseudo_data=pseudo_data, name_suffix=PSEUDO_MODEL_SUFFIX.format(iter=(i+1)),
+                           **kwargs)
+            current_score = self.info()['best_model_score_val']
+
+            logger.log(20,
+                       f'Pseudolabeling algorithm changed validation score from: {previous_score}, to: {current_score}'
+                       f' using evaluation metric: {self.eval_metric.name}')
+
+            if previous_score >= current_score:
+                break
+            else:
+                # Cut down X_test to not include pseudo labeled data
+                X_test = X_test.loc[test_pseudo_idxes[~test_pseudo_idxes].index]
+                previous_score = current_score
+
+        if return_pred_prob:
+            return self, y_pred_proba_og
+        else:
+            return self
+
+    def fit_pseudolabel(self, test_data: pd.DataFrame, max_iter: int = 5, return_pred_prob: bool = False, **kwargs):
+        """
+        Pseudolabeling fit algorithm. If test_data is labeled then incorporates all
+        test_data into train_data for newly fit models. This data is not used for
+        validation. If test_data is unlabeled then fit_pseudolabel will self label the
+        test data and incorporate all the test data that meets a criteria (For example
```

Comment:
"incorporate all the test data" is vague. can clarify as: "will augment the original training data by adding all the pseudolabeled test data"